src/Adapter/Cart/CommandHandler/UpdateCartCarrierHandler.php

```diff
@@ -71,6 +71,7 @@ public function handle(UpdateCartCarrierCommand $command)
             ->setCustomer(new Customer($cart->id_customer))
             ->setShop(new Shop($cart->id_shop))
         ;
+        $this->contextStateManager->saveCurrentContext();
```

Comment:
You should probably use the helper function `setCartContext` It's probably not in this `AbstractCartHandler` but it's worth adding it in the abstract I think